### ISSUE TITLE
Replace Jabo's Audio with Azimers Audio

### ISSUE THREAD

**Issue by project64** [project64/project64#433]
since @Azimer now has open sourced his plugin (https://github.com/Azimer/AziAudio) It would be nice to include this in project64 to replaced the closed source plugin by jabo. 

What do you think? If it is done, how do you think it is best to be done, my thoughts have been to:
* Fork it and stick it in the main line similar to nrage (that way building project64 with installer, will have it there). I can then also make sure it uses the same property sheets and compiled with the same version of Visual studio.
* Clone it and have it as a separate project under the project64 account with the upstream as the original, I would most likely then have to add the compiled binary in to the project64 branch so that when the installer is built/zip package then we have the binary there.
* Look at some other method to add it in to the build process (sub modules, is there something like composer for c++?)

Any other good ideas?

**Comment by cxd4** [comment]
Practical convenience isn't everything.  You may find it more challenging to keep components to Project64 separate to their own repositories, but what's easy isn't the same thing always as what's right.

This is what Git repositories were born for.  It's how the cycle works:  Each repository can have a fork with mutual communication back and forth for directly and easily merging changes from upstream/downstream.  You happen to not have any immediate value for this feature with N-Rage because N-Rage has no maintainer now that rabiddeity has left.

libretro-mupen64plus RetroArch fork is a counterexample because our fork of Mupen64Plus doesn't use plugin-based design, so it's not possible for us to clone plugin repositories directly because libretro cores statically link everything to a single core with all the CPU, RCP stuff etc. all done in the same library.  You however are using a plugin system.  You have more than enough ability to rely on organization to separate repositories, but you believe it's too difficult because of how the installer is integrated for public releases of Project64.

**Comment by AmbientMalice** [comment]
I fully support replacing Jabo with Azimer's. Any issues that arise can very likely be resolved, wheras Jabo's is unfortunately a development dead end.

I personally think the project should be forked into PJ64, and any changes to Azimer's main version can be added to PJ64's fork every week or so. But I'm not a git expert.

**Comment by cxd4** [comment]
I can say this much off hand, if it is forked into the same repository as PJ64 along with NRage and everything else my RAM lockup issues with opening the entire solution in VS2013 Community won't get any better.

**Comment by oddMLan** [comment]
I agree, Azimer's plugin should be forked in a separate repo, not integrated in the already bloated MSVC solution.

**Comment by Lithium64** [comment]
To have it integrated into the project64 any option is welcome

**Comment by Frank-74** [comment]
I'd like it included into Project64.  But there would need to be changes to get it to build in VS2008.  All the C99 includes would need replacing.

**Comment by cxd4** [comment]
There is I think stdint.h ... what else?

I'm already more than able to take care of the stdint.h part as I don't practice it anyway.

Also not sure how well / if Azimer's XAudio can link with VS2008.

**Comment by project64** [comment]
@cxd4 I got it compiling with vs2008 https://github.com/project64/AziAudio tho I moved files and changed things from c to c++ .. 

will break the vs2013 and the build you made for gcc, still got to fix that up ... not sure if @Azimer  would want this structural change as a pull request

**Comment by Frank-74** [comment]
@project64 Just built using your AziAudio.vs2008.sln.  I get 169 warnings and build succeeds with XA2.  But DX version fails.
When I build with Azimers sln using codeblocks and VS2008 compiler, I get no warnings and build succeeds with DX version.  But XA2 fails with link error, but I think thats down to not linking ole32.lib.
https://github.com/Azimer/AziAudio/issues/29

**Comment by Azimer** [comment]
If you can fix up what you can, I'd be willing to deal with the structural changes.  There are a lot of structural changes going on anyway.  It's good to get them out of the way before we really start digging into HLE and playback bug resolution.

**Comment by cxd4** [comment]
I don't care that they were renamed from .c to .cpp file extension; they are still C code and I have no plans to modify any of bobby smiles' files as of yet.  Therefore the friction between Azimer/AziAudio and project64/AziAudio is rather minimal, if only because of the MusyX C stuff changed to CPP file ext.

>will break the vs2013 and the build you made for gcc, still got to fix that up

Nah, just calls for simple renaming of the source files and that should work again.

**Comment by cxd4** [comment]
I made a pull request to fix the GCC build but forgot to clone/fork the separate repo under pj64's account.  I realized in the process however that it is a problem that all of Azimer's source is under Azimer/AziAudio/AziAudio and pj64 under project64/AziAudio/Source/AziAudio.  That totally conflicts the directory structure for being able to directly fetch/merge changes between the repositories.

The "3rd Party" folder also is a slight problem (though one which may be ignored perhaps).  MinGW is still Unix-based and should not have spaces in paths, so I had to fix the pj64 MinGW build by changing the include path to $src\..\3RDPAR~1\directx\include.

So the solutions I could think of are:
* Azimer could move `AziAudio/` to `Source/AziAudio`, to match what zilmar is doing.
* pj64 could move `Source/AziAudio` and `Source/3rd Party` to the root of the repo.

Either solution seems equally stable though I tend to prefer not doing the former because then it breaks the commit history for file tracking changes to Azimer's repository, whereas project64's AziAudio repository is brand new and has basically no commit history to lose.  So I would tend to prefer the latter.

**Comment by project64** [comment]
> @project64 Just built using your AziAudio.vs2008.sln. I get 169 warnings and build succeeds with XA2. But DX version fails.

the warnings are that I use warning level 4 and I assume you are on warning level 3

I need to see why the audio is broken with it.

> whereas project64's AziAudio repository is brand new and has basically no commit history to lose

the files should be moved so should have a history, there should be a complete history there and if it is not then it broken, and I need to do it again.

**Comment by cxd4** [comment]
The commit history is still there, but it's not viewable directly when you view a file on GitHub and click the "History" or "Blame" buttons.  It will show you all the commits as far back as the file's creation under that name...if there was a commit before that where the name of the file was something different it would get treated as an entirely different file.  In other words you have to go as far back as possible and change the history view for a different file each time it was renamed.

I don't see the source for concern.  Why is having a "Source" folder in the root necessary?  I can understand if it's just to be consistent with the `project64` repository which does have a Source folder in it, but since you're currently exercising the freedom of not yet having both repositories merged into one I don't see how that's important at the moment.  Besides, GitHub was meant for source code, so, a folder named "Source" sounds somewhat redundant.

**Comment by Frank-74** [comment]
> the warnings are that I use warning level 4 and I assume you are on warning level 3.
> I need to see why the audio is broken with it.
Audio is not broken with XA2 version. it works fine here in Project64.  DS8 doesn't build though.

**Comment by cxd4** [comment]
Dunno what's being discussed here but XA2 and DS8 work fine in the VS2008 Express pull request I just filed.

**Comment by cxd4** [comment]
>Clone it and have it as a separate project under the project64 account with the upstream as the original, I would most likely then have to add the compiled binary in to the project64 branch so that when the installer is built/zip package then we have the binary there.

Well I'm happy you went with this solution!  I hope that in the long run, you'll feel the same way.

Sure it might be a little hard to know for sure since you haven't done any public releases of PJ in a while.

**Comment by oddMLan** [comment]
Should be closed since Project64-Audio based on Azimer's is the default now.
